test/examples/ucp_client_server.c

```diff
@@ -0,0 +1,450 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+/*
+ * UCP client - server example utility
+ * -----------------------------------------------
+ *
+ * Server side:
+ *
+ *    UCX_NET_DEVICES=mlx5_0:1 UCX_TLS=rc ./ucp_client_server
+ *
+ * Client side:
+ *
+ *    UCX_NET_DEVICES=mlx5_0:1 UCX_TLS=rc ./ucp_client_server <server-ip>
+ *
+ * Notes:
+ *
+ *    - The server will listen to incoming connection requests on INADDR_ANY.
+ *    - The client needs to pass the IP address of the server side to connect to,
+ *      as the first and only argument to the test.
+ *    - Currently, the passed IP needs to be an IPoIB address.
+ *    - The amount of used resources (HCA's and transports) needs to be limited
+ *      for this test (for example: UCX_NET_DEVICES=mlx5_0:1 UCX_TLS=rc_x).
+ *      This is currently required since the UCP layer has a limitation on
+ *      the size of the transfered transports addresses that are being passed
+ *      to the remote peer.
+ *
+ */
+
+#include "ucx_hello_world.h"
+
+#include <ucp/api/ucp.h>
+
+#include <arpa/inet.h> /*inet_addr */
+#include <unistd.h>  /* getopt */
+#include <ctype.h>   /* isprint */
+#include <pthread.h> /* pthread_self */
```

Comment:
we don't really need to use pthread_self in the test

Reply:
@yosefe it is used in the print in the stream_recv_cb and stream_send_cb functions. you think it's better to repmove those?

Reply:
yes, this is not needed in this example